Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIDeclarativeNetRequestCocoa.mm

```diff
@@ -222,6 +222,46 @@
     completionHandler(std::nullopt);
 }
 
+void WebExtensionContext::declarativeNetRequestGetMatchedRules(std::optional<WebExtensionTabIdentifier> tabIdentifier, std::optional<double> minTimeStamp, CompletionHandler<void(std::optional<Vector<WebExtensionMatchedRuleParameters>> matchedRules, std::optional<String>)>&& completionHandler)
+{
+    RefPtr tab = tabIdentifier ? getTab(tabIdentifier.value()) : nil;
+
+    if (tabIdentifier && !tab) {
+        completionHandler(std::nullopt, toErrorString(@"declarativeNetRequest.getMatchedRules()", @"tabId", @"The specified tab does not exist."));
```

Comment:
`toErrorString(@"declarativeNetRequest.getMatchedRules()", nil, @"tab not found")` to match other API errors.